Feature/sched 26

stuemke opened the pull request:
new feature to record job executions.

appng-buildmaster committed:
tagging release 1.11.2

appng-buildmaster committed:
tagging release 1.11.3

appng-buildmaster committed:
tagging release 1.11.3

Commit:
Merge tag 'appng-scheduler-1.11.3'

appng-buildmaster committed:
Merge branch 'appng-scheduler-1.11.x'

appng-buildmaster committed:
Merge branch 'appng-scheduler-1.11.x'

appng-buildmaster committed:
Merge branch 'appng-scheduler-1.11.x'

Commit:
SCHED-26 add feature to record and visualize job executions

Commit:
generate message constants, add datasource and filters on separate tab

Commit:
add REST API for job records

Commit:
add test, refactor bean configuration, add doku

Commit:
some refactoring based on static code analysis

Commit:
Add license header, author and comments

madness-inc commented on src/main/java/org/appng/application/scheduler/job/JobRecordHouseKeepingJob.java:
```diff
@@ -0,0 +1,83 @@
+ * 
+ * @author Claus Stümke
+ *
+ */
```
Why not use lombok's @Data?

stuemke replied:
done. see latest commit

madness-inc commented on src/main/java/org/appng/application/scheduler/service/RecordsRestController.java:
```diff
@@ -0,0 +1,84 @@
+@RestController
+public class RecordsRestController {
+
+	@Autowired
```
use constructor injection, makes getter/setter obsolete

madness-inc commented on src/main/java/org/appng/application/scheduler/model/JobResult.java:
```diff
@@ -0,0 +1,87 @@
+import org.appng.api.ScheduledJobResult.ExecutionResult;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
```
Wyh not use @Data?

stuemke replied:
done. see latest commit. But you have to be carefule using @ Annotations in comments. Maybe you can change the one above?

madness-inc commented on src/main/java/org/appng/application/scheduler/model/JobRecord.java:
```diff
@@ -0,0 +1,70 @@
+ * @author Claus Stümke
+ *
+ */
+
```
Why not use @Data?

madness-inc commented on application-home/beans.xml:
```diff
@@ -79,11 +95,32 @@
+		<property name="jobDataMap">
+			<map>
+				<entry key="enabled" value="${houseKeepingEnabled}" />
+				<entry key="cronExpression" value="${indexExpression}" />
```
indexExpression is used for lucene indexing. Use dedicated property?

madness-inc commented on src/main/java/org/appng/application/scheduler/business/Records.java:
```diff
@@ -0,0 +1,123 @@
+ * @author Claus Stümke
+ *
+ */
+public class Records implements DataProvider {
```
Define as @Component and use constructor injection (remove from beans.xml)

stuemke replied:
done. see latest commit

madness-inc commented on src/main/java/org/appng/application/scheduler/business/SchedulingController.java:
```diff
@@ -36,16 +37,20 @@
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Pageable;
 
 public class SchedulingController extends SchedulerAware implements ApplicationController {
```
Define as @Component and use constructor injection (remove from beans.xml)

stuemke replied:
done. see latest commit

madness-inc commented on src/main/java/org/appng/application/scheduler/business/Records.java:
```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2011-2017 the original author or authors.
```
must be 2011-2018

madness-inc reviewed (changes requested):
see comments

Commit:
Refactoring after feedback. Use component annotations instead of bean
declarations. Use lombok instead of boilderplate code. Use distict
properties for new house keeping job.

madness-inc commented on src/main/java/org/appng/application/scheduler/service/JobRecordService.java:
```diff
@@ -0,0 +1,197 @@
+
+	}
+
+	private boolean addFiler(String filterName, String operator, String filterValue, StringBuilder sql,
```
Shoud be addFil**t**er

madness-inc commented on src/main/java/org/appng/application/scheduler/service/JobRecordService.java:
```diff
@@ -0,0 +1,197 @@
+		return addFiler(filterName, operator, filterValue, sql, paramsMap, first, filterName);
+	}
+
+	private boolean addFiler(String filterName, String operator, String filterValue, StringBuilder sql,
```
Shoud be addFil**t**er

stuemke replied:
Typo has been corrected

madness-inc commented on src/main/java/org/appng/application/scheduler/business/Records.java:
```diff
@@ -0,0 +1,130 @@
+		filter.getSelections().add(startedBeforeFilter);
+	}
+
+	public JobRecordService getJobRecordService() {
```
Getter/setter not need due to constructor injection

stuemke replied:
getter and setter have been removed

Commit:
Fix typo and remove not needed getter and setter

Commit:
add tests and job record details action

madness-inc commented on pom.xml:
```diff
@@ -156,6 +173,47 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
```
Use appNG Maven Plugin instead
https://github.com/appNG/appng/tree/appng-1.17.1/appng-maven-plugin

stuemke replied:
Done. Using appng-maven-plugin now

Commit:
use appng maven plugin to generate constants classes

madness-inc reviewed (approved):
Great work!

madness-inc reviewed (approved):
Great Work!
